tt_metal/hw/inc/risc_common.h

```diff
@@ -191,4 +191,18 @@ inline __attribute__((always_inline)) void conditionally_disable_l1_cache() {
 #endif
 }
 
+// Flush i$ on ethernet riscs
+inline __attribute__((always_inline)) void flush_erisc_icache() {
+#ifdef ARCH_BLACKHOLE
+    // Kernel start instructions on WH are not cached because we apply a 1 cache line (32B) padding
+    //  between FW end and Kernel start.
+    // This works because risc tries to prefetch 1 cache line.
+    // The 32B still get cached but they are never executed
+    #pragma GCC unroll 2048
+    for (int i = 0; i < 2048; i++) {
```

Comment:
I think I asked this before...how big is the cache on ierisc?  This loop is sized for 8K which seems large, but maybe my memory is just failing

Reply:
icache on eth was increased from 512B to 8KB. We don't have explicit i$ flush on WH because there is a pad applied via linker scripts that prevents kernels from inadvertently getting cached.